File: web/models/conversation-file.ts
export type ConversationFilePayload = {
  schema_version: 1
  file_id: string
  name: string
  size: number
  mime_type: string
  width?: number | null
  height?: number | null
  thumbnail_file_id?: string | null
  hash?: string | null
}

export type ConversationFileUploadResponse = ConversationFilePayload & {
  access_url: string
}

export type ConversationFileAccessResponse = {
  url: string
  expires_seconds: number
}

File: server/app/schemas/conversation_file.py
"""
Schemas for conversation file upload and access.
"""
from pydantic import BaseModel, Field


class ConversationFileUploadResponse(BaseModel):
    schema_version: int = 1
    file_id: str = Field(..., min_length=1)
    name: str
    size: int
    mime_type: str
    access_url: str


class ConversationFileAccessResponse(BaseModel):
    url: str
    expires_seconds: int
